### ISSUE TITLE
Errors in posterior mean when evaluating batches of points

### ISSUE THREAD

**Issue by pabloprf** [pytorch/botorch#549]
When evaluating the mean of the posterior of a model, is there any difference between evaluating a single point (dimensions `1 x dim`) versus a a batch of points (dimensions `batch x dim`)? 

Below I include a working case that gives me a small error of the order of `1E-8` when evaluating one of the training points once v.s. twice. This is a small error, but in my real-world problem, this kind of error propagates and becomes larger when I perform transformations to the input space when I build custom models.

I wonder if there is something that I am doing wrong or whether this difference is expected. If it is the latter, is there a way to increase accuracy? Naively, I would expect the evaluation of the posterior mean to be deterministic, but I do not know enough about the code.

## Code example

This is a minimum working example of a case with a single output and 8 input dimensions:

```
import torch
import numpy as np
from botorch.models.gp_regression import FixedNoiseGP
from gpytorch.mlls import ExactMarginalLogLikelihood
from botorch.fit import fit_gpytorch_model
from botorch.optim.fit import fit_gpytorch_torch


train_X = torch.Tensor([[0.5000, 0.5000, 0.5000, 0.5000, 0.5000, 0.5000, 0.5000, 0.5000],
        [0.4909, 0.3459, 0.6945, 0.7175, 0.7447, 0.1615, 0.8161, 0.2229],
        [0.5051, 0.7082, 0.1507, 0.3822, 0.3920, 0.4814, 0.6154, 0.6951],
        [0.1372, 0.9100, 0.7858, 0.9862, 0.1059, 0.8537, 0.2678, 0.2718],
        [0.7796, 0.1788, 0.4479, 0.1362, 0.8805, 0.6998, 0.1094, 0.9436]])

train_Y = torch.Tensor([[0.1710],
        [0.0070],
        [0.7466],
        [0.0000],
        [1.0000]])

train_Yvar = torch.Tensor([[1.0000e-04],
        [1.0000e-04],
        [1.0000e-04],
        [1.0000e-04],
        [1.0000e-04]])

model 	= FixedNoiseGP(train_X, train_Y,train_Yvar)
mll 	= ExactMarginalLogLikelihood(model.likelihood, model)

mll,f 	= fit_gpytorch_torch(mll)


with torch.no_grad():

	x0 = train_X[0].unsqueeze(0)

	y1 = model.posterior(x0).mean[0].item()

	y2 = model.posterior(np.repeat(x0,2,axis=0)).mean[0].item()
	print('\t\t--> Error (2 samples v.s. 1): {0}'.format(np.abs(y1-y2)))

```

Output:
```
		--> Error (2 samples v.s. 1): 1.4901161193847656e-08
```

## System Info
Please provide information about your setup, including
- BoTorch Version 0.3.0
- GPyTorch Version 1.1.1
- PyTorch Version 1.6.0
- Computer OS: macOS Catalina

**Comment by Balandat** [comment]
This looks ok to me, it's a really tiny discrepancy. Posterior inference means solving a linear system, in one case it's just a scalar, in the other case a 2x2 matrix. Mathematically there is no difference, my suspicion is that this is just numerical precision issues (possibly b/c of implicitly using different solvers under the hood)
